docs/cloud/pouta/tutorials/image-pipeline.md

```diff
@@ -0,0 +1,650 @@
+
+$ source allas_conf --mode S3 --user your-csc-username
+```
+
+Once these three steps are done, we test that everything works properly by listing our buckets in Allas:
+```
+$ s3cmd ls
+2025-01-16 14:59  s3://input_bucket
+2025-01-16 14:59  s3://output_bucket
+```
+We notice that we can see the two buckets that we have created earlier in the tutorial.
+We have thus confirmed that the virtual machine can now access properly the buckets in Allas.
+
+### Configuring access from virtual machine to database
+
+At this point in the tutorial, traffic from the virtual machine is allowed to flow towards the database in Pukki.
+However, at the moment the virtual machine has no information about where the database can be found, i.e., its public IP address, nor which are the credentials to use when accessing the database.
+Therefore, next we configure the access to the database from the virtual machine.
+
+We stay on `terminal_C`.
+First, we install a tool that we will need to talk with the database:
+```
+$ sudo apt install postgresql-client
+Reading package lists... Done
+Building dependency tree... Done
+Reading state information... Done
+The following NEW packages will be installed:
+  postgresql-client
+0 upgraded, 1 newly installed, 0 to remove and 28 not upgraded.
+Need to get 11.6 kB of archives.
+After this operation, 17.4 kB of additional disk space will be used.
+Get:1 http://nova.clouds.archive.ubuntu.com/ubuntu noble-updates/main amd64 postgresql-client all 16+257build1.1 [11.6 kB]
+Fetched 11.6 kB in 0s (47.3 kB/s)             
+Selecting previously unselected package postgresql-client.
+(Reading database ... 117863 files and directories currently installed.)
+Preparing to unpack .../postgresql-client_16+257build1.1_all.deb ...
+Unpacking postgresql-client (16+257build1.1) ...
+Setting up postgresql-client (16+257build1.1) ...
+Scanning processes...                                                                                                                                                                    
+Scanning linux images...                                                                                                                                                                 
+
+Running kernel seems to be up-to-date.
+
+No services need to be restarted.
+
+No containers need to be restarted.
+
+No user sessions are running outdated binaries.
+
+No VM guests are running outdated hypervisor (qemu) binaries on this host.
+```
+
+We then prepare a script with the information on how to reach and access the database hosted in Pukki.
+On `terminal_C` we type the following:
+```
+$ nano db_cred.sh
+```
+We see a blinking cursor on the top-left corner of the window, indicating that the terminal is ready to receive our input.
+We write in the terminal what follows:
+```
+#!/bin/bash
+export DB_HOST="5.6.7.8"
+export DB_PORT="5432"
+export DB_NAME="pipeline_db"
+export DB_USER="db_admin"
+export PGPASSWORD='db_password'
+```
```

Comment:
Postgres has its own config file that is loaded automatically. Maybe it is a better and easier approach?

Reply:
Fixed in https://github.com/CSCfi/csc-user-guide/pull/2424/commits/9b1b28a56261b814d7262c0afb27932b6881381e